### ISSUE TITLE
Export with upscaling turned on - sometimes it fails

### ISSUE THREAD

**Issue by PDeb108** [darktable-org/darktable#16819]
### Describe the bug

Export fails for some photographs when 'allow upscaling' is set to 'yes'.  It is not a consistent failure for every photo as, for some, it exports without any problem.  When it fails it creates a zero byte image in the destination.  However, the export is successful if the 'allow upscaling' Is set to no

This is a recent problem.  I did not encounter this in a few versions before in DT 4.7 .  I cannot recall the version number when it was OK

Here is the export config that is causing problems

![image](https://github.com/darktable-org/darktable/assets/165793573/5ccce57a-0119-4c65-8778-fc044071a707)

### Steps to reproduce

![image](https://github.com/darktable-org/darktable/assets/165793573/2fc8ba46-e45f-4e59-976d-07ad21a747f2)

It fails in this export config

### Expected behavior

It should export without any issues

### Logfile | Screenshot | Screencast

[darktable-log.txt](https://github.com/darktable-org/darktable/files/15367446/darktable-log.txt)

### Where did you obtain darktable from?

GitHub nightly

### darktable version

4.7.0+1243~ged3f06eead

### What OS are you using?

Windows

### What is the version of your OS?

Version 11

### Describe your system?

Acer Swift SF314-511

Processor	11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz   2.42 GHz
Installed RAM	8.00 GB (7.80 GB usable)
Device ID	8B27435A-8086-467F-A43E-1CDD681952B3
Product ID	00356-24525-53642-AAOEM
System type	64-bit operating system, x64-based processor
Pen and touch	No pen or touch input is available for this display

Edition	Windows 11 Home Single Language
Version	23H2
Installed on	‎02-‎10-‎2022
OS build	22631.3593
Experience	Windows Feature Experience Pack 1000.22700.1003.0

### Are you using OpenCL GPU in darktable?

I dont know

### If yes, what is the GPU card and driver?

Intel(R) Iris(R) Xe Graphics

**Comment by PDeb108** [comment]
added an offending file and the associated sidecar file

**Comment by gi-man** [comment]
PD, I tried to reproduce this on Windows 11 using Bill weekly from today. I used your same image/xmp and same export settings and it always exported in CPU or GPU path. I also tried in Fedora 40 KDE.

The logfile uploaded seems like it had a typo in the `darktable -d pipe -d imageio`. Can you create a new log?

Do you have Lua scripts in your system?

**Comment by PDeb108** [comment]
Hi, Attaching a new log file.  I have already updated to the latest nightly build darktable-4.7.0+1245.g1a6d1ab1b1-win64, before I saw your message.  The problem remains.  Yes I have Lua scripts in my system

[darktable-log.txt](https://github.com/darktable-org/darktable/files/15371704/darktable-log.txt)

**Comment by PDeb108** [comment]
If this is of any help.
The problem is ONLY with all the photos I download from Pixls.us-play raw, not when I import the photos from my camera.  The play raw downloads I keep in a separate folder and these do not follow the import naming pattern.  But  I have seen this problem only lately, it wasn't so two weeks back with the nightly build.  If I could have a nightly build that is more than two weeks old, I could verify my statement

**Comment by gi-man** [comment]
@jenshannoschwalm I was able to reproduce the issue this morning on todays master on F40 KDE (self build gcc14.1.1) https://pastebin.com/F67QaVdR

But, it is weird. I tested with another image, same export settings, no issue. I went back to the OP image, same issues. I started turning off sigmoid module, issue went away. Turned it back on, no issue. Multiple attempts, no issue.

Turned off dt and restarted. Exported from lighttable = issue, open in darkroom = no issue. Switched all modules off, restarted/export from lighttable = issue.

**Comment by PDeb108** [comment]
You are better off than I am.  Tried to export from Darkroom=issue.  Export with sigmoid off=issue.  At least you have recreated the problem, which gives hope a solution will be found..  It behaves better in a Linux environment it seems.   Thanks so far

**Comment by PDeb108** [comment]
I have an interesting development.  I installed the latest version - Darktable-4.7.0+1271.gf98eda41f0-win64, and the problem has disappeared, or I should say, none of the photos that were causing problems earlier, exported perfectly.  FYI

**Comment by gi-man** [comment]
There is another PR (https://github.com/darktable-org/darktable/pull/16840) not yet merged that should address this problem. Keep testing on your system once it is merged.

Not related to this subject, I noticed you have a GPU card, but it is disabled in your darktablerc file. The edits on your image are CPU intensive. Turning on your GPU should give you a performance gain.

**Comment by PDeb108** [comment]
Will continue to check it out with the latest nightly build every day. 
GPU card enable - Oh, how do I enable it? As I am not technically competent, I will need step-by-step advice, if you can.  Thanks

**Comment by gi-man** [comment]
The manual explains it. You have to edit a string by changing a 0/1. Check/ask on pixls.us since it is outside the scope of this bug.

**Comment by gi-man** [comment]
I tested on master (darktable 4.7.0+1278~g0e47c61f9c) and I confirm no longer getting this issue.
